internal/tx/txnprocessor.go

```diff
@@ -133,6 +150,25 @@ func (p *txnProcessor) Init(rpc eth.RPCClient) {
 	if p.conf.HDWalletConf.URLTemplate != "" {
 		p.hdwallet = newHDWallet(&p.conf.HDWalletConf)
 	}
+	p.concurrencySlots = make(chan bool, p.conf.SendConcurrency)
```

Comment:
After a discussion with @peterbroadhurst , I note that this needs to be setup in `Init()` because of the way this function and the caller of this function `NewKafkaBridge()` is called: https://github.com/hyperledger/firefly-ethconnect/blob/442cfb353a6e6f265a358bef2e39c07e0e1f2c0d/cmd/ethconnect.go#L172-L173

The config is set _after_ the constructor is invoked, so setting it in the constructor doesn't have the intended effect.